relay_test.go

```diff
@@ -516,6 +516,62 @@ func TestRelayMakeOutgoingCall(t *testing.T) {
 	})
 }
 
+func TestRelayInboundConnContext(t *testing.T) {
+	rh := relaytest.NewStubRelayHost()
+	rh.SetFrameFn(func(f relay.CallFrame, conn *relay.Conn) {
+		// Verify that the relay gets the base context set in ConnContext
+		assert.Equal(t, "bar", conn.Context.Value("foo"), "Unexpected value set in base context")
+	})
+	opts := testutils.NewOpts().SetRelayOnly().SetRelayHost(rh).SetConnContext(func(ctx context.Context, conn net.Conn) context.Context {
+		return context.WithValue(ctx, "foo", "bar")
+	})
+
+	testutils.WithTestServer(t, opts, func(t testing.TB, ts *testutils.TestServer) {
+		rly := ts.Relay()
+		svrOpts := testutils.NewOpts().SetServiceName("svc").SetConnContext(func(c context.Context, conn net.Conn) context.Context {
+			return context.WithValue(c, "foo", "bar")
+		})
+		svr := ts.NewServer(svrOpts)
+		testutils.RegisterFunc(svr, "echo", func(ctx context.Context, args *raw.Args) (*raw.Res, error) {
+			// Verify that the context passed into the handler inherits from the base context
+			// set by ConnContext
+			assert.Equal(t, "bar", ctx.Value("foo"), "Value unexpectedly different from base context")
+			return &raw.Res{Arg2: args.Arg2, Arg3: args.Arg3}, nil
+		})
+
+		err := testutils.CallEcho(rly, ts.HostPort(), "svc", nil)
+		assert.NoError(t, err, "Echo failed")
+	})
+}
+
+func TestRelayContextInheritsFromOutboundConnection(t *testing.T) {
+	rh := relaytest.NewStubRelayHost()
+	rh.SetFrameFn(func(f relay.CallFrame, conn *relay.Conn) {
+		assert.Equal(t, "bar", conn.Context.Value("foo"), "Unexpected value set in base context")
+	})
+	opts := testutils.NewOpts().SetRelayOnly().SetRelayHost(rh)
+
+	testutils.WithTestServer(t, opts, func(t testing.TB, ts *testutils.TestServer) {
+		rly := ts.Relay()
+		calleeOpts := testutils.NewOpts().SetServiceName("svc2")
+		callee := ts.NewServer(calleeOpts)
+		testutils.RegisterEcho(callee, nil)
```

Comment:
i think we can do `callee := ts.Server()` since a server is created by default when using `WithTestServer`. it'll have a default service name, but you can access that using `ts.Server().ServiceName()` or maybe also `ts.ServiceName()`